liblumen_alloc/src/erts/term/map.rs

```diff
@@ -39,6 +39,12 @@ impl Map {
         Self::from_hash_map(value)
     }
 
+    pub fn put(&self, key: Term, value: Term) -> HashMap<Term, Term> {
+        let mut map = self.value.clone();
+        map.insert(key, value);
```

Comment:
This could be more memory efficient (although slower because of the double probe for the key) if it detected if the key already had the value and did not create a clone in such a case.  This version is correct though, so we can optimize in later PR.

Reply:
I recon cloning a large map could easily take more time than the two checks.